napari/layers/image/experimental/octree_image.py

```diff
@@ -429,3 +453,29 @@ def on_chunk_loaded(self, request: ChunkRequest) -> None:
         if self._slice.on_chunk_loaded(request):
             # Tell the visual to redraw with this new chunk.
             self.events.loaded()
+
+    @property
+    def delay_ms(self) -> NormalNoise:
+        """Return the currently configured artificial load delay.
+
+        Return
+        ------
+        NormalNoise
+            The current configured delay.
+        """
+        return self._delay_ms
+
+    @delay_ms.setter
+    def delay_ms(self, delay_ms: NormalNoise):
+        """Set the new artificial load delay.
+
+        We sometimes want to simulate latency for debugging or demos.
+
+        Parameters
+        ----------
+        delay_ms : NormalNoise
+            Optional delay to simulate latency.
+        """
+        self._delay_ms = delay_ms
+        self._slice = None  # For now must explicitly delete it
+        self.refresh()  # Create a new slice.
```

Comment:
I love the delay adding for demo purposed, but am I right in thinking that in the long run we don't want these delays living on the main `OctreeImage` object to avoid things getting cluttered?

Reply:
Yeah it shouldn't be there long term. I'm not sure where I'd put it right now. It is a per-layer setting though.

In general `OctreeImage` is kind of following the pattern of `Image` and becoming just kind of a kitchen sink of stuff without a lot of structure. A big refactoring might want to until there is someday a single image class again.

Reply:
ok that's fine. i'll merge this as is, but agreed we'll want to think more about this long term